tests/core/test_basics.py

```diff
@@ -139,6 +139,40 @@ def expected_response_type() -> Type[DummyResponse]:
     assert stream_counter == 2
 
 
+@pytest.mark.asyncio
+async def test_stream_can_get_cancelled(endpoint: Endpoint) -> None:
+    stream_counter = 0
+
+    async_generator = endpoint.stream(DummyRequest)
+
+    async def stream_response() -> None:
+        async for event in async_generator:
+            # Accessing `ev.property_of_dummy_request` here allows us to validate
+            # mypy has the type information we think it has. We run mypy on the tests.
+            print(event.property_of_dummy_request)
+            nonlocal stream_counter
+            stream_counter += 1
+            await asyncio.sleep(0.5)
+
+    async def cancel_soon() -> None:
+        while True:
+            await asyncio.sleep(0.01)
+            if stream_counter == 2:
+                await async_generator.aclose()
```

Comment:
@pipermerriam I did some research and I believe this is the default way to cancel an `AsyncGenerator`. See: https://www.python.org/dev/peps/pep-0525/#id14

Reply:
Related: It looks like we should change out tear down code everywhere were event loops are used to look like this:

```python
try:
    loop.run_forever()
finally:
    loop.run_until_complete(loop.shutdown_asyncgens())
    loop.close()
```
See: https://docs.python.org/3/library/asyncio-eventloop.html?highlight=loop#asyncio.loop.shutdown_asyncgens

Reply:
Open an issue in py-evm codebase for this?